cli/src/cli_util.rs

```diff
@@ -2866,6 +2866,27 @@ pub fn print_unmatched_explicit_paths<'a>(
     Ok(())
 }
 
+#[cfg(feature = "git2")]
+pub fn print_git2_deprecation_warning(
+    ui: &Ui,
+    workspace_command: &WorkspaceCommandHelper,
+) -> Result<(), CommandError> {
+    let settings = workspace_command.settings();
+    if !settings.git_settings()?.subprocess
+        && !settings
+            .get("debug.suppress-git2-deprecation-warning")
+            .optional()?
+            .unwrap_or(false)
+    {
+        writeln!(
+            ui.warning_default(),
+            "`git.subprocess = false` will be removed in 0.30; please report any issues you have \
+             with the default.",
+        )?;
+    }
+    Ok(())
+}
```

Comment:
nit: might be simpler to move this function to `git_util`, and handle `cfg(feature = "git2")` internally, but I don't have strong feeling.

Reply:
I somewhat like the `#[cfg]` because it makes it easier to see what’s due for removal. `git_util` doesn’t depend on `cli_util` (which contains `WorkspaceCommandHelper`) currently (but `cli_util` does depend on `git_util`), do you think breaking that property to do the move would be okay?

Reply:
I think `settings` can be passed in to this function instead. I tend to avoid inserting `cfg(..)` at random places, but that's just my preference.

Reply:
I moved it and adjusted the signature. I agree about `#[cfg]` in general, but I think in this case it helps make removal a little more mechanical, so I left it for now.